### DISCUSSION TITLE
Where to write code for advanced playlist

### DISCUSSION THREAD

**Discussion by DJMic** [AzuraCast/AzuraCast#6856] [Q&A]
v0.19.3 Stable
Docker.

Hi,
I'm wondering where to write custom code for advanced playlist.
Into the documentation it says this:

![image](https://github.com/AzuraCast/AzuraCast/assets/156111736/915c46a8-09a7-41ee-bcbc-a1a0d3f120e4)

ButI don't find this section to write input code 

![image](https://github.com/AzuraCast/AzuraCast/assets/156111736/0539ce98-da8d-4e5f-b4ba-487256d412d0)

Could you tell me where to find it ?

**Comment by gAlleb** [comment]
Hi! 
`Station Profile` - `Broadcasting` - `Edit Liquidsoap Configuration`

<img width="500" alt="Снимок экрана 2024-01-09 в 20 13 07" src="https://github.com/AzuraCast/AzuraCast/assets/144447266/bd663d45-0e6c-4743-8b9f-122b54268e5a">

P.S. Perhaps this should be changed in docs :)

**Comment by DJMic** [comment]
Thanks for your answer.
Another question : Where is the best place to write code ?
Before declaring the playlists as below  ? 
![image](https://github.com/AzuraCast/AzuraCast/assets/156111736/9f9253ca-3e0b-4b87-9314-318681ec70f4)

Or in this place 

![image](https://github.com/AzuraCast/AzuraCast/assets/156111736/ff00b991-0a29-4540-a32e-d2c907b8f3a4)

Which is not ideal when using the advanced playlist configuration, Azuracast requires you to restart the station. Is there a workaround to still be able to write into this file without having to reload the station ?

**Comment by gAlleb** [comment]
> Where is the best place to write code ?

It depends on what you want to add to the script.

> Is there a workaround to still be able to write into this file without having to reload the station ?

No. Liquidsoap has to be restarted in order to apply changes to configuration.

**Reply by DJMic** [reply]
However, when you edit standard playlists (change Random, Once every x songs, ...) , modifications are directly written into the liquidsoap configuration file, without reloading the station. But not when using advanced playlists :(

**Reply by gAlleb** [reply]
You can `Reload Configuration` as per 
```
Stations using Icecast can soft-reload the station configuration, applying changes while keeping the stream broadcast running.
Reloading broadcasting will not disconnect your listeners.
```
But again this will restart Liquidsoap anyway.

> However, when you edit standard playlists (change Random, Once every x songs, ...) , modifications are directly written into the liquidsoap configuration file, without reloading the station. But not when using advanced playlists :(

No, they are not. They would be written only after restart (you can see changed configuration — but it is not really changed yet and if you open `Logs` - `Liquidsoap Configuration` it would confirm this). The reason those modifications work without reloading is because Azuracast manages and handles those changes. Some functions and changes require that Liquidsoap be reloaded. Some not.
